components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java

```diff
@@ -424,6 +424,48 @@ public void revokeOAuthConsentByApplicationAndUser(String username, String tenan
         }
     }
 
+    /**
+     * Revoke the OAuth consents by the application and authorized users's tenant domain which are recorded in the
+     * IDN_OPENID_USER_RPS table.
+     *
+     * @param applicationName Name of the OAuth App
+     * @param tenantDomains   Tenant domains of authorized users (Non SaaS applications accommodate only one tenant
+     *                        in the array).
+     * @throws IdentityOAuth2Exception If an unexpected error occurs.
+     */
+    @Override
+    public void revokeOAuthConsentsByApplication(String applicationName, String[] tenantDomains)
+            throws IdentityOAuth2Exception {
+
+        for (String tenantDomain : tenantDomains) {
+            if (log.isDebugEnabled()) {
+                log.debug("Revoking OAuth consent for application: " + applicationName
+                        + " in tenant: " + tenantDomain);
+            }
+
+            Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+            PreparedStatement ps = null;
+
+            try {
+                String sql = SQLQueries.DELETE_USER_RPS_OF_APPLICATION;
+
+                ps = connection.prepareStatement(sql);
+                ps.setInt(1, IdentityTenantUtil.getTenantId(tenantDomain));
```

Comment:
Shall we have this resolved in advance?
`IdentityTenantUtil.getTenantId(tenantDomain)`